src/libraries/Common/src/Extensions/ActivatorUtilities/ActivatorUtilities.cs

```diff
@@ -49,11 +52,9 @@ public static object CreateInstance(IServiceProvider provider, Type instanceType
 
             if (!instanceType.GetTypeInfo().IsAbstract)
             {
-                foreach (ConstructorInfo? constructor in instanceType
-                    .GetTypeInfo()
-                    .DeclaredConstructors)
+                foreach (ConstructorInfo? constructor in instanceType.GetConstructors())
                 {
-                    if (!constructor.IsStatic && constructor.IsPublic)
```

Comment:
Doesn't this change the runtime behavior?

Reply:
```C#
foreach (ConstructorInfo? constructor in instanceType.GetConstructors())
```

That will only return `public` constructors.
What we were doing before was getting ALL constructors, but then filtering just the public constructors.
```C#
                foreach (ConstructorInfo? constructor in instanceType
                    .GetTypeInfo()
                    .DeclaredConstructors)
                {
                    if (!constructor.IsStatic && constructor.IsPublic)
                    {
                    }
                }
```

Reply:
Got it, GetConstructors returns public constructors only.